src/Workspaces/Core/Portable/Rename/ConflictEngine/ConflictResolver.Session.cs

```diff
@@ -851,6 +857,40 @@ private bool ShouldIncludeLocation(ISet<RenameLocation> renameLocations, RenameL
 
                 return RenameLocation.ShouldRename(location);
             }
+
+            /// <summary>
+            /// We try to compute the sub-spans to rename within the given <paramref name="containingLocationForStringOrComment"/>.
+            /// If we are renaming within a string, the locations to rename are always within this containing string location
+            /// and we can identify these sub-spans.
+            /// However, if we are renaming within a comment, the rename locations can be anywhere in trivia,
+            /// so we return null and the rename rewriter will perform a complete regex match within comment trivia
+            /// and rename all matches instead of specific matches.
+            /// </summary>
+            private static ImmutableSortedSet<TextSpan>? GetSubSpansToRenameInStringAndCommentTextSpans(
+                TextSpan containingLocationForStringOrComment,
+                IEnumerable<RenameLocation> locationsToRename)
+            {
+                var builder = ImmutableSortedSet.CreateBuilder<TextSpan>();
+                foreach (var renameLocation in locationsToRename)
+                {
+                    if (!containingLocationForStringOrComment.Contains(renameLocation.Location.SourceSpan))
```

Comment:
this can happen? how?

Reply:
This happens when renaming in comments. The current logic for renaming in comment trivia is putting the token's span (not full span), so it does not include the actual trivia span which contains the actual locations to rename. The rename rewriter is hooked up to only look at token spans when finding things to rename, and then explicitly looks in trivia to find rename locations in comments. We can potentially cleanup the logic in comments renaming to use proper containing spans, but I did not want to touch that in this PR.